pkg/kubeadm/master/addons.go

```diff
@@ -223,11 +230,15 @@ func createKubeDNSServiceSpec(s *kubeadmapi.KubeadmConfig) (*api.ServiceSpec, er
 }
 
 func CreateEssentialAddons(s *kubeadmapi.KubeadmConfig, client *clientset.Clientset) error {
-	kubeProxyDaemonSet := NewDaemonSet(kubeProxy, createKubeProxyPodSpec(s))
-	SetMasterTaintTolerations(&kubeProxyDaemonSet.Spec.Template.ObjectMeta)
+	arches := [3]string{"amd64", "arm", "arm64"}
+
+	for _, arch := range arches {
```

Comment:
Why not having it this way from the start?
This is daemonset specific, since we can't select specific images based on arch, so three different daemonsets have to be created. Only one of them will be used in most cases anyway, so it doesn't cost anything.